### PR TITLE
Explicit note that tmpdir fixture is discouraged in favour of tmp_path #9937

### LINKED ISSUE #9937: Explicit note that tmpdir fixture is discouraged in favour of tmp_path
While https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures` mentions "These days, prefer to use tmp_path and tmp_path_factory", it's quite hidden in the text.

I propose to announce this more loudly using a note admonition. Additionally the note admonition should be placed in the docstrings:

- https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=tmpdir#std-fixture-tmpdir
- https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=tmpdir#pytest.legacypath.LegacyTmpdirPlugin.tmpdir

The same holds for `tmpdir_factory`.

### PR DESCRIPTION
**Pull request by Wolfremium13**
Explicit note that ``tmpdir`` fixture is discouraged in favour of ``tmp_path``.

Closes #9937 

- [N/A] Include documentation when adding new features.
- [N/A] Include new tests or update existing tests when applicable.
- [X] Allow maintainers to push and squash when merging my commits. Please uncheck this if you prefer to squash the commits yourself.
- [X] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/main/changelog/README.rst) for details.
- [X] Add yourself to `AUTHORS` in alphabetical order.

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -87,6 +87,7 @@ Damian Skrzypczak
 Daniel Grana
 Daniel Hahler
 Daniel Nuri
+Daniel Sánchez Castelló
```
Comment (nicoddemus) [inline AUTHORS:90]: Why are you adding this many people? Usually PR authors just add themselves. Not sure it is OK to add other people without their consent.

File: changelog/9937.doc.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Explicit note that ``tmpdir`` fixture is discouraged in favour of ``tmp_path``.
```
Comment (nicoddemus) [inline changelog/9937.doc.rst:1]:
```suggestion
Explicit note that :fixture:`tmpdir` fixture is discouraged in favour of :fixture:`tmp_path`.
```

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Thanks @Wolfremium13, appreciate it!

Please take a look at my comments.

File: AUTHORS (hunk 1 above)
Comment (DaniSancas) [reply AUTHORS:90]: Hi @nicoddemus, this PR is part of a mob-programming session at @clarityai-eng.

The people involved in the thinking and writing of this change are:
@toclarity - Tatiana
@Wolfremium13 - Kevin
@victorrmclarity - Victor
@DaniSancas - Daniel

Is it correct to write all our names in the AUTHORS section or should we delete some of them?

Comment (nicoddemus) [reply AUTHORS:90]: Ahh no worries then, thanks for the clarification.

Comment (DaniSancas) [reply AUTHORS:90]: Yes, everyone on the list was in the session and gave their consent.

It's the first time we contribute to this project. Are there any next steps to be done?

Thank you!

**Review by nicoddemus — APPROVED** [review]

File: AUTHORS (hunk 1 above)
Comment (nicoddemus) [reply AUTHORS:90]: No that's it!

Thanks @toclarity, @Wolfremium13, @victorrmclarity, and @DaniSancas for the contribution! 👍

**Pull request merged**